ui/components/Header/index.tsx

```diff
@@ -102,13 +104,31 @@ const Header = ({
 }) => {
   const router = useRouter();
   const [isMenuOpen, setMenuOpen] = useState(false);
+  const [userFromStorage, setUserFromStorage] = useState(null);
 
   const [, joinGroup] = useMutation(JOIN_GROUP_MUTATION);
   const [, acceptInvitation] = useMutation(ACCEPT_INVITATION);
 
   const [, joinRound] = useMutation(JOIN_ROUND_MUTATION);
   const color = round?.color ?? "anthracit";
 
+  useEffect(() => {
+    if (currentUser) {
+      set("user", currentUser);
+    }
+  }, [currentUser]);
+
+  useEffect(() => {
+    const user = get("user");
+    if (user && fetchingUser) {
+      setUserFromStorage(user);
+    }
+    else if (!fetchingUser && currentUser === null) {
+      remove("user");
+      setUserFromStorage(null);
+    }
+  }, [fetchingUser, currentUser]);
+
```

Comment:
Instead of storing the user in local storage, would it be possible to decode the cookie already saved instead?